ts/ui/menu/AnnotationMenu.ts

```diff
@@ -0,0 +1,188 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2019-2022 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  The annotations submenus.
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ * @author v.sorge@mathjax.org (Volker Sorge)
+ */
+
+import {Submenu} from 'mj-context-menu/js/item_submenu.js';
+import {MJContextMenu} from './MJContextMenu.js';
+import {MmlNode} from '../../core/MmlTree/MmlNode.js';
+import {SelectableInfo} from './SelectableInfo.js';
+import * as MenuUtil from './MenuUtil.js';
+
+/**
+ * The annotation types to look for in a MathItem. These are options set in the
+ * Menu object.
+ */
+let annotationTypes: {[type: string]: string[]} = {};
+
+/**
+ * Sets the connected annotation box.
+ * @param {SelectableInfo} box The info box.
+ */
+export function setAnnotationBox(box: SelectableInfo) {
+  annotationBox = box;
+}
+
+/**
+ * The info box for showing annotations (created by the Menu object that
+ * contains this MJContextMenu)
+ */
+let annotationBox: SelectableInfo = null;
```

Comment:
I'm a little leary about making these values global rather than tied to a context instance like the `MJContextMenu` instance.  It is possible that there are two separate MathDocuments (say one each for two frames in a window) and so two separate contexts where annotations could be used.  I've tried to avoid globals like this and make everything be tied to an instance so that MathJax can have multiple separate contexts at once.  That's one reason these were part of `MJContextMenu` originally.

Reply:
I've removed all globals now and even caching is handled by a handshake Array.